### ISSUE TITLE
WordPress block has large whitespace at the bottom

### ISSUE THREAD

**Issue by notnownikki** [WordPress/gutenberg#14659]
**Describe the bug**

In the editor, inserting a WordPress block and embedding a post renders with a load of whitespace. It seems this happened because the html returned for embedding a post changed at some point.

cc @jasmussen

**Comment by jasmussen** [comment]
Confirmed. In fact it's not just any random sized space, the block is being treated as "responsive 16:9".

Editor:

<img width="831" alt="Screenshot 2019-03-28 at 09 51 43" src="https://user-images.githubusercontent.com/1204802/55143547-4261b580-513f-11e9-9cea-2418d49e5564.png">

Frontend:

<img width="864" alt="Screenshot 2019-03-28 at 09 51 49" src="https://user-images.githubusercontent.com/1204802/55143551-442b7900-513f-11e9-9d4d-9f49d9f6ffe3.png">

Actually... hmmmmmmmm. I think maybe the embed block is "fine", because the bug as shown above is actually because the theme I'm testing with is providing its own responsive embed styles, which should not be there.

In your testing environment, what theme did you use and which plugins did you have enabled?

It's possible that if a regression happened, it happened in the responsive-embeds opt-in function call — maybe it opts in for all themes now even when it shouldn't? Not at all sure, but right at this point my gut tells me this isn't a Gutenberg issue.

**Comment by notnownikki** [comment]
Stock WP no plugins, 2019 theme. (Afk today, sorry for the shortness of
email!)

**Comment by jasmussen** [comment]
@kjellr can you recall if TwentyNineteen opts into responsive embeds? Or does it have its own embed responsiveness magic?

**Comment by kjellr** [comment]
Yes, Twenty Nineteen does opt in to responsive embeds.

**Comment by notnownikki** [comment]
When I'm back I'll check if the block does any responsive stuff. If it does
it and the embed is now coming inside an iframe, it should opt out. I know
the html did change for WP embeds, so it's possible that's what's going on.

**Comment by notnownikki** [comment]
Checking this out, and found...

![embeddedWP](https://user-images.githubusercontent.com/3314354/55221746-2246fa80-5202-11e9-8a00-83a04a7d37c6.png)


```
	{
		name: 'core-embed/wordpress',
		settings: {
			title: 'WordPress',
			icon: embedWordPressIcon,
			keywords: [ __( 'post' ), __( 'blog' ) ],
			responsive: false,
			description: __( 'Embed a WordPress post.' ),
		},
	},
```

The WordPress block does _not_ have responsive styles applied, in the editor or the front end, and yet there's still a load of padding at the bottom. The `iframe` that the embed is rendered in has the padding...

![embeddedWP](https://user-images.githubusercontent.com/3314354/55221598-bd8ba000-5201-11e9-9b79-b0686959ae3d.png)

That iframe is coming from `oembed`, and there are no responsive styles applied to the wrapping `figure`.

**Comment by jasmussen** [comment]
That padding _is_ the responsive style, it's 56.something% right? That's the responsive style for a 16:9 video. Which obviously this isn't. This is freeform, so it should not have any of that. And I'm not sure where it's coming from, because that padding should only come from the 16:9 CSS classes

**Comment by swissspidy** [comment]
Worth noting that WordPress post embeds (iframes) calculate their height and pass that information to the embedding site (host) via `postMessage`. That way, the host can re-adjust the height. There should be no need to do anything else except setting a width of 100% for the WordPress embed iframe.

**Comment by notnownikki** [comment]
@jasmussen no, there are no responsive classes in the embedded html at all. Unless they're being applied to all iframes (which would be a horrible theme bug) then they're not applied there. Also, the dimensions shown there are the actual `height` and `width` attributes of the iframe in the html. It's not responsive styles making that size.

**Comment by swissspidy** [comment]
Worth noting that WordPress embed iframes have a minimum height of 200px. See https://github.com/WordPress/wordpress-develop/blob/2a7aedc9e08a4a3701f908ffc981ac5a476b493c/src/js/_enqueues/wp/embed.js#L62-L72

@notnownikki This might be what you are running into at the moment.

**Comment by notnownikki** [comment]
This is the HTML `oembed` generates:

```
<iframe class="wp-embedded-content" sandbox="allow-scripts" security="restricted" src="http://localhost:8888/?p=1&amp;embed=true#?secret=U0rOggtHDr" data-secret="U0rOggtHDr" width="600" height="200" title="“Hello world!” — Gutenberg Dev" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

```

It's fixed height, and it's including the page as the source of the iframe, and if I remember correctly this is quite different to what used to happen. Now the iframe is a fixed height, which means shorter content going to have that white space, as the iframe does not resize.

In the editor and in the core blocks, there are no responsive styles applied. This looks like it's down to an oembed change which also meant we needed to change the WordPress HTML detection in https://github.com/WordPress/gutenberg/pull/14658 (which needs a review ;) )

**Comment by swissspidy** [comment]
> Now the iframe is a fixed height, which means shorter content going to have that white space, as the iframe does not resize.

It **should** resize using postMessage calls. That's what `wp-embed.js` is for.

**Comment by notnownikki** [comment]
Ah. It's not doing that here. For a post that simple says "hello world!" the content in the iframe is 600x132, but the containing iframe remains 600x200.

**Comment by swissspidy** [comment]
Can you verify whether that `wp-embed.js` JS file is being enqueued on your site and that the iframe has a `data-secret` attribute? I'll try to reproduce on my end in the meantime.

**Comment by notnownikki** [comment]
Yes, it's enqueued (the `.min` version) and there is a `data-secret` attribute in the iframe. https://github.com/WordPress/gutenberg/issues/14659#issuecomment-477944512 has the HTML that appears in the post.

**Comment by jasmussen** [comment]
I can't really reproduce in master + 2019:

![embeds](https://user-images.githubusercontent.com/1204802/55227910-fe42e380-5218-11e9-9a36-4fc6d9cb10a2.gif)

What am I missing?

I can reproduce in [my own theme](https://github.com/jasmussen/navi) because I haven't updated it in a while, and it provides its own heavy-handed responsive styles.

**Comment by notnownikki** [comment]
I think you _are_ seeing it there, can you inspect the iframe on the shorter embed there?, the "Learn something difficult" one? I'm pretty sure I can see the bottom whitespace when you highlight it. If the enclosing iframe is still 200px high, you've reproduced it.

**Comment by notnownikki** [comment]
> I mean, there's a LITTLE bit of extra padding below the "short" embed, but nothing that looks super wrong.

Yeah, that's a symptom. The problem is when the iframe is 1000px high, it _should_ resize, so when a post is _just_ enough to tip it over 200px, it'll go to 1000px and you'll see the massive whitespace.

If you inspect the short embed and you see the wrapping iframe is still 200px, then it hasn't resized and that's the problem.

**Comment by jasmussen** [comment]
Yep, I see it now. There's definitely that 200px min-height that Pascal is referring to. Which would be nice to fix, but is also not horrible since it doesn't crop anything.

But I still can't reproduce the latter thing you're referring to — do you have a link to a post that results in a 1000px embed?

**Comment by notnownikki** [comment]
Only on my local machine. This is how embedding the default "Welcome to WordPress" post looks in my browser. I can't guarantee that it would render pixel-perfect identical for you though, because of different browsers, OSes, etc. and so it _might_ not trigger the problem for you.

![1000px](https://user-images.githubusercontent.com/3314354/55234443-13247480-5223-11e9-917e-da24b1aaf499.png)

**Comment by notnownikki** [comment]
If I remember we had a similar issue with the resizing iframe in Gutenberg, and it was that the initial messages posted weren't caught by the resizing listener, because of $REASON (I think it was some release of Chrome which broke it?) and so we had to fire off our own message when we registered the listener.l

**Comment by jasmussen** [comment]
That's interesting, I can repeat it now:

![Screenshot 2019-03-29 at 14 05 49](https://user-images.githubusercontent.com/1204802/55234604-cf823880-522b-11e9-8668-fd40d01c900d.png)

That page is from the same dev site. It smells race-condition-y, without knowing virtually ANYTHING about the embeds (so don't take my word). In this case the iframe is 338px tall. Note it looks okayish on the frontend where it's 200px tall:

![Screenshot 2019-03-29 at 14 08 05](https://user-images.githubusercontent.com/1204802/55234750-1a9c4b80-522c-11e9-9e3e-54eadb6cfa80.png)

Given that this is a freeform iframe where the height is actually the attribute set on the iframe, I'm not sure I can be of that much help here :( 

![Screenshot 2019-03-29 at 14 08 34](https://user-images.githubusercontent.com/1204802/55234801-3869b080-522c-11e9-927d-d3bde6527811.png)

**Comment by notnownikki** [comment]
Whew, ok, it's not just me :grin: 

@swissspidy if your resizing code isn't firing, it might be that it's missing the first message. We fixed it in the sandbox component by running an initial resize ourselves, because we were missing the very first resize message after we registered the listener. https://github.com/WordPress/gutenberg/blob/master/packages/components/src/sandbox/index.js#L131

**Comment by annezazu** [comment]
I can still replicate this with WordPress 6.6.1 and GB 19.1:

![Screenshot 2024-09-05 at 8 34 07 PM](https://github.com/user-attachments/assets/455e0d13-a028-4c4a-91be-12fc49db5f17)

**Comment by jasmussen** [comment]
The issue appears to not be the responsive embeds feature anymore. In my testing, it appears more like a race condition or something the like. Specifically it appears the explicit height is set on the iframe, on the frontend only, to match a calculated height:

![Screenshot 2024-09-06 at 09 29 00](https://github.com/user-attachments/assets/a4497bdf-2de8-4fdb-9e69-1df032c86116)

But that calculation must just be wrong. @ellatrix when you get a moment, can you look?

**Comment by devansh016** [comment]
<img width="1469" alt="Image" src="https://github.com/user-attachments/assets/5afb70de-cbb2-4d85-8de7-5a612e341d6b" />

This seems to be problem due to `height="200"` attribute in iframe. I tried to change this value but it changes back to 200 instantly.

**Comment by devansh016** [comment]
I was able to replicate this issue on WordPress develop repo also. This issue should be created on core tracker, this is not Gutenberg specific issue. Since the code which is causing this issue is on WordPress develop repo.

Core Ticket: https://core.trac.wordpress.org/ticket/62934

**Comment by carolinan** [comment]
Note that the core ticket linked above has been closed as wontfix.

Pages that can be embedded for testing:
https://wordpress.org/blocks/ taller than 200px, so there are no spacing issues

https://make.wordpress.org/core/handbook/tutorials/trac/ short page with spacing issue.

**Comment by jasmussen** [comment]
There has been little activity on this issue since it was opened, so perhaps its impact is limited. In respect of the developers closing as wontfix on the core tracker, I'm going to go ahead and do the same here, but with the note that if you feel this is in error, please comment on the core issue instead. OR, if this closing was in error, LMK and we can reopen!
